Raven.Database/Raft/ClusterManager.cs

```diff
@@ -30,10 +35,89 @@ public class ClusterManager : IDisposable
 
         public bool HasNonEmptyTopology => Engine.CurrentTopology.AllNodes.Any();
 
-        public ClusterManager(RaftEngine engine)
+        private Timer timer = null;
+
+        public ClusterManager(RaftEngine engine, DatabasesLandlord databasesLandlord)
         {
             Engine = engine;
             Client = new ClusterManagementHttpClient(engine);
+            DatabasesLandlord = databasesLandlord;
+            engine.StateChanged += OnRaftEngineStateChanged;
+        }
+
+        private DatabasesLandlord DatabasesLandlord { get; set; }
+
+        private void OnRaftEngineStateChanged(RaftEngineState state)
+        {
+            if (state == RaftEngineState.Leader)
+            {
+                var period = DatabasesLandlord.SystemDatabase.Configuration.Cluster.MaxReplicationLatency;
+                //timer should be null 
+                timer?.Dispose();
+                timer = new Timer(NewLeaderTimerCallback,null, period, period); 
+            }
+            else
+            {
+                //If this node was the leader before, dispose of the timer.
+                timer?.Dispose();
+            }
+        }
+
+        private void NewLeaderTimerCallback(object state)
+        {
+            //This method should never be invoked from non-leader
+            //Might happen if the timer and the state change event happen at the same time.
+            if (Engine.State != RaftEngineState.Leader)
+            {
+                if(Log.IsDebugEnabled)
+                    Log.Debug($"NewLeaderTimerCallback invoked from non-leader node, actual state:{Engine.State}");
+                return;
+            }
+            var databasesNames = GetDatabasesNames();
+            var databaseToLastModified = new Dictionary<string,Tuple<DateTime,string>>();
+            foreach (var databaseName in databasesNames)
+            {
+                try
+                {
+                    Task<DocumentDatabase> databaseTask;
+                    if (DatabasesLandlord.TryGetOrCreateResourceStore(databaseName, out databaseTask) == false)
+                        continue;
+                    var database = databaseTask.Result;
```

Comment:
That isn't the issue, the issue is that you are waiting for the db to load, and that shouldn't be here at all